tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+			}
+		}
+	}
+}
+
+func unmountRootPartition(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.Label != "ROOT" {
+			continue
+		}
+
+		umountOut, err := exec.Command(
+			"/bin/umount", partition.Device).CombinedOutput()
+		if err != nil {
+			t.Fatal("umount", err, string(umountOut))
+		}
+	}
+}
+
+func unmountPartitions(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.FilesystemType == "" || partition.Label == "ROOT" {
+			continue
+		}
+		umountOut, err := exec.Command(
+			"/bin/umount", partition.Device).CombinedOutput()
+		if err != nil {
+			t.Fatal("umount", err, string(umountOut))
+		}
+	}
+}
+
+func setExpectedPartitionsDrive(actual []*Partition, expected []*Partition) {
+	for _, a := range actual {
+		for _, e := range expected {
+			if a.Number == e.Number {
+				e.MountPath = a.MountPath
+				e.Device = a.Device
+				break
+			}
+		}
+	}
+}
+
+func validatePartitions(
+	t *testing.T, expected []*Partition, fileName string) {
+	for _, e := range expected {
+		if e.TypeCode == "blank" {
+			continue
+		}
+		sgdiskInfo, err := exec.Command(
+			"/sbin/sgdisk", "-i", strconv.Itoa(e.Number),
+			fileName).CombinedOutput()
+		if err != nil {
+			t.Fatal("sgdisk -i", strconv.Itoa(e.Number), err)
+		}
+		lines := strings.Split(string(sgdiskInfo), "\n")
+		actualTypeGUID := strings.ToUpper(strings.TrimSpace(
+			strings.Split(strings.Split(lines[0], ": ")[1], " ")[0]))
+		actualSectors := strings.Split(strings.Split(lines[4], ": ")[1], " ")[0]
+		actualLabel := strings.Split(strings.Split(lines[6], ": ")[1], "'")[1]
+
+		// have to align the size to the nearest sector first
+		expectedSectors := align(e.Length, 512)
+
+		if e.TypeGUID != actualTypeGUID {
+			t.Fatal("TypeGUID does not match!", e.TypeGUID, actualTypeGUID)
+		}
+		if e.Label != actualLabel {
+			t.Fatal("Label does not match!", e.Label, actualLabel)
+		}
+		if strconv.Itoa(expectedSectors) != actualSectors {
+			t.Fatal(
+				"Sectors does not match!", expectedSectors, actualSectors)
+		}
+
+		if e.FilesystemType == "" {
+			continue
+		}
+
+		df, err := exec.Command("/bin/df", "-T", e.Device).CombinedOutput()
+		if err != nil {
+			t.Fatal("df -T", err, string(df))
+		}
+
+		lines = strings.Split(string(df), "\n")
+		if len(lines) < 2 {
+			t.Fatal("Couldn't verify FilesystemType")
+		}
+		actualFilesystemType := removeEmpty(strings.Split(lines[1], " "))[1]
+
+		if e.FilesystemType != actualFilesystemType {
+			t.Fatal("FilesystemType does not match!", e.Label,
+				e.FilesystemType, actualFilesystemType)
+		}
+	}
+}
+
+func validateFiles(t *testing.T, expected []*Partition) {
+	for _, partition := range expected {
+		if partition.Files == nil {
+			continue
+		}
+		for _, file := range partition.Files {
+			path := filepath.Join(partition.MountPath, file.Path, file.Name)
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				t.Fatal("File doesn't exist!", path)
```

Comment:
If you call `t.Error` instead of `t.Fatal` in places like this, when something goes wrong the tester will be able to see all the errors instead of just the first one that was found.